include/cocaine/hpack/header.hpp

```diff
@@ -269,18 +210,7 @@ class header_table_t {
     find(const std::function<bool(const header_t&)> comp);
 
     // Header storage. Implemented as circular buffer
-    std::array<header_t, max_header_capacity> headers;
-    size_t header_lower_bound;
-    size_t header_upper_bound;
-
-    // Header data storage. Stores all data which headers can reference.
-    // Implemented as a sort of circular buffer.
-    // We multiply by 2 as data can be padded and we don't want to move it in memory.
-    // 2 multiplier guarantee that we can add new value to the end or beginning without data overlap.
-    std::array<char, max_data_capacity*2> header_data;
-    size_t data_lower_bound;
-    size_t data_lower_bound_end;
-    size_t data_upper_bound;
+    std::deque<header_t> headers;
```

Comment:
it IS. But as far as we need to copy data during unpacking I don't see any need of old data disposition.